### PR TITLE
Katello 4.14 release notes

### PR DESCRIPTION
**Pull request by chris1984**
#### What changes are you introducing?

Katello 4.14 Release notes, contibutors, headline features and upgrade warnings

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

https://community.theforeman.org/t/katello-4-14-rc1-release-procedure/39182

#### Checklists

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

### CODE REVIEW

File: guides/doc-Release_Notes/master.adoc [hunk 1]
```diff
@@ -2,7 +2,7 @@ include::common/attributes.adoc[]
 include::common/header.adoc[]
 
 ifndef::katello[]
-include::topics/foreman.adoc[]
+include::topics/katello-4.14.0.adoc[leveloffset=+1]
```
Comment (ekohl) [inline guides/doc-Release_Notes/master.adoc:5]: Please insert this at the right place. There's an explicit instruction for this in https://community.theforeman.org/t/katello-4-14-rc1-release-procedure/39182 that does exactly that:
> `sed '/x.y.z releases here/a include::topics/katello-4.14.0.adoc[leveloffset=+1]' master.adoc`

File: guides/doc-Release_Notes/topics/katello-4.14.0.adoc [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+* pass:[Tests failing on Ruby 2.7 due to sorting of content types in the message] - https://projects.theforeman.org/issues/37681[#37681]
+* pass:[Multiple environments can be assigned to a host even if setting should prevent it] - https://projects.theforeman.org/issues/37657[#37657]
+* pass:[Smart Proxy referred to as "proxy" in settings] - https://projects.theforeman.org/issues/37656[#37656]
+* pass:[Use custom snippets during Errata Installation] - https://projects.theforeman.org/issues/37654[#37654]
+* pass:[Clean up defunct deb content mirror_publication_options] - https://projects.theforeman.org/issues/37595[#37595]
+* pass:["Upload profile - Katello Script Default" doesn't work for SLES] - https://projects.theforeman.org/issues/37569[#37569]
+* pass:[Angular tests failing on master-source] - https://projects.theforeman.org/issues/37563[#37563]
+* pass:[sudo katello-tracer-upload fails] - https://projects.theforeman.org/issues/37561[#37561]
+* pass:[Host and hostgroup form shows multiple Library entries] - https://projects.theforeman.org/issues/37174[#37174]
```
Comment (ekohl) [inline guides/doc-Release_Notes/topics/katello-4.14.0.adoc:16]: Not blocking, but ideally these would be categorized in Redmine.

File: guides/doc-Release_Notes/topics/katello-4.14.0.adoc [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+* pass:[Host Collections widget should respect the errata_status_installable setting] - https://projects.theforeman.org/issues/37288[#37288]
+
+=== Hosts
+
+* pass:[Content source is changed to wrong proxy if you simply press the submit button on the host page] - https://projects.theforeman.org/issues/37709[#37709]
+* pass:[Host update failure: param is missing or the value is empty: content_facet_attributes] - https://projects.theforeman.org/issues/37704[#37704]
+* pass:[Convert2rhel env facts are getting filtered due to env ethernet regex] - https://projects.theforeman.org/issues/37696[#37696]
+* pass:['TypeError: getCustomizedRexUrl is not a function' when you go to Review step in Packages wizard] - https://projects.theforeman.org/issues/37684[#37684]
+* pass:[Add postgresql-evr as a migration] - https://projects.theforeman.org/issues/37678[#37678]
```
Comment (ekohl) [inline guides/doc-Release_Notes/topics/katello-4.14.0.adoc:59]: I think this was reverted and shouldn't be in the changelog. I reopened the issue.

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 4]
```diff
@@ -4,12 +4,49 @@
 [id="katello-headline-features"]
 == Headline Features
 
-There are no highlights with Katello {KatelloVersion}.
+=== New All Hosts index page improvements ===
+
+The experimental new All Hosts index page gets some major improvements with this release. To use the new page by default, in Settings, set Show new host overview page to Yes.
```
Comment (ekohl) [inline guides/doc-Release_Notes/topics/katello.adoc:9]: We put sentences on their own lines. Quoting https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md#pull-request-checklist
> I put each sentence to its own line.

See https://sembr.org/ for more.

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 5]
```diff
@@ -4,12 +4,49 @@
...
+
+*Host bulk actions* - You can now select one or more hosts and click the vertical ellipsis menu to perform the following actions:
+Manage content
+Packages (install / remove / upgrade)
+Errata
+Content source
+Content view environments - change the content view and lifecycle environment of hosts
```
Comment (ekohl) [inline guides/doc-Release_Notes/topics/katello.adoc:16]: Should this be a bullet list?
```suggestion
* Manage content
* Packages (install / remove / upgrade)
* Errata
* Content source
* Content view environments - change the content view and lifecycle environment of hosts
```

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 6]
```diff
@@ -4,12 +4,49 @@
 [id="katello-headline-features"]
 == Headline Features
 
-There are no highlights with Katello {KatelloVersion}.
+=== New All Hosts index page improvements ===
```
Comment (ekohl) [inline guides/doc-Release_Notes/topics/katello.adoc:7]:
```suggestion
=== New All Hosts index page improvements
```

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 7]
```diff
@@ -4,12 +4,49 @@
...
+Packages (install / remove / upgrade)
+Errata
+Content source
+Content view environments - change the content view and lifecycle environment of hosts
+
+*Single-host actions* - In addition to actions added by Foreman, you can click the vertical ellipsis in the rightmost table column (without selecting the host) to change content view environments on a single host.
+
+*Column selector* - You can now click Manage Columns and customize columns displayed in the table. The column options are the same as those on the legacy All Hosts page. Any changes made here will also reflect on the legacy All Hosts page, and vice versa.
+
+=== Multiple Content Views ===
```
Comment (ekohl) [inline guides/doc-Release_Notes/topics/katello.adoc:22]:
```suggestion
=== Multiple Content Views
```

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 8]
```diff
@@ -4,12 +4,49 @@
...
+
+Assigning multiple content view environments to hosts
```
Comment (ekohl) [inline guides/doc-Release_Notes/topics/katello.adoc:24]: This feels like a header.

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 9]
```diff
@@ -4,12 +4,49 @@
...
+
+Assigning or registering a host to multiple environments requires the Allow multiple content views Setting to be on. If you turn this setting off, existing multi-environment hosts will continue to work, but you will not be able to assign multiple environments to hosts going forward.
+
+You can also assign multiple environments at registration using subscription-manager register --environments, but global registration and activation keys are not yet supported.
 
 [id="katello-upgrade-warnings"]
 == Upgrade Warnings
 
-There are no upgrade warnings with Katello {KatelloVersion}.
+Extra Pulpcore indexes to address a https://github.com/pulp/pulpcore/issues/5367[performance issue] will be added during the upgrade. If you've manually created these indexes, please remove them before proceeding with the upgrade to ensure that the migration runs successfully.
```
Comment (ekohl) [inline guides/doc-Release_Notes/topics/katello.adoc:47]: Please give this its own header

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 10]
```diff
@@ -4,12 +4,49 @@
...
+On the new All Hosts page, there is a new 'Content view environments' column available to add. This will display all assigned environments for a host. The existing 'Content view' and 'Lifecycle environment' columns will continue to work, but will only display the first assigned environment.
+
+If you reassign a host's content view environments using the web UI, the host will no longer be a multi-environment host.
+
+Assigning or registering a host to multiple environments requires the Allow multiple content views Setting to be on. If you turn this setting off, existing multi-environment hosts will continue to work, but you will not be able to assign multiple environments to hosts going forward.
+
+You can also assign multiple environments at registration using subscription-manager register --environments, but global registration and activation keys are not yet supported.
 
 [id="katello-upgrade-warnings"]
 == Upgrade Warnings
```
Comment (ekohl) [inline guides/doc-Release_Notes/topics/katello.adoc:45]: https://github.com/theforeman/foreman-documentation/pull/3180 should also be included. If @ehelms doesn't finish it then you should take over the effort.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: guides/doc-Release_Notes/topics/katello.adoc (hunk 10 above)
Comment (chris1984) [reply guides/doc-Release_Notes/topics/katello.adoc:45]: @ehelms it looks like they want you to rebase on the 3.12 branch. https://github.com/theforeman/foreman-documentation/pull/3180#pullrequestreview-2254314673

Did you want me to just take your note and add this to mine?

**Review by chris1984 — COMMENTED** [review]
@ekohl updated all the items, even categorized the Redmine issues. The only thing I have not touched yet is the release note from Eric, I left him a comment on the thread where you pinged him to ask if he wants me to add it here.

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 11]
```diff
@@ -4,12 +4,55 @@
 [id="katello-headline-features"]
 == Headline Features
 
-There are no highlights with Katello {KatelloVersion}.
+=== New All Hosts index page improvements
+
+The experimental new All Hosts index page gets some major improvements with this release.
+To use the new page by default, in Settings, set Show new host overview page to Yes.
+
+*Host bulk actions* - You can now select one or more hosts and click the vertical ellipsis menu to perform the following actions:
```
Comment (bangelic) [inline guides/doc-Release_Notes/topics/katello.adoc:12]:
```suggestion
*Host bulk actions* - You can now select one or more hosts and click the vertical ellipsis menu to perform the following actions:

```

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 12]
```diff
@@ -4,12 +4,55 @@
...
+* Packages (install / remove / upgrade)
+* Errata
+* Content source
+* Content view environments - change the content view and lifecycle environment of hosts
+
+*Single-host actions* - In addition to actions added by Foreman, you can click the vertical ellipsis in the rightmost table column (without selecting the host) to change content view environments on a single host.
+
+*Column selector* - You can now click Manage Columns and customize columns displayed in the table.
+The column options are the same as those on the legacy All Hosts page. 
+Any changes made here will also reflect on the legacy All Hosts page, and vice versa.
```
Comment (bangelic) [inline guides/doc-Release_Notes/topics/katello.adoc:23]:
```suggestion
Any changes made in either the new experimental *All Hosts* page and the legacy *All Hosts* page reflect on each other.
```

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 13]
```diff
@@ -4,12 +4,55 @@
...
+
+=== Assigning multiple content view environments to hosts
+
+You can now assign multiple content view environments to a host via subscription-manager or Hammer. A multi-environment host will have access to enabled repositories from all of its assigned content view environments. This strategy may be more convenient than composite content views for instances where frequent publishes are required.
```
Comment (bangelic) [inline guides/doc-Release_Notes/topics/katello.adoc:27]:
```suggestion
You can now assign multiple content view environments to a host through subscription-manager or Hammer. A multi-environment host will have access to enabled repositories from all of its assigned content view environments. This strategy may be more convenient than composite content views for instances where frequent publishes are required.
```

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 14]
```diff
@@ -4,12 +4,55 @@
...
+
+Via hammer, pass the `content-view-environments` param to `hammer host update` in a comma-separated list in the format `<lifecycle_environment_label>/<content_view_label>` . Ex `hammer host update --id xxx --content-view-environments lce1/cv1,lce2/cv2`
```
Comment (bangelic) [inline guides/doc-Release_Notes/topics/katello.adoc:29]:
```suggestion
Through hammer, pass the `content-view-environments` param to `hammer host update` in a comma-separated list in the format `<lifecycle_environment_label>/<content_view_label>` . Ex `hammer host update --id xxx --content-view-environments lce1/cv1,lce2/cv2`
```

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 15]
```diff
@@ -4,12 +4,55 @@
...
+To see the currently assigned content view environment(s), use subscription-manager environments --list-enabled
+
+The web UI does not yet support assigning multiple environments. 
+However, multi-environment hosts will now be displayed in the web UI on the new All Hosts page and new host details page.
+
+On the new All Hosts page, there is a new 'Content view environments' column available to add. This will display all assigned environments for a host. The existing 'Content view' and 'Lifecycle environment' columns will continue to work, but will only display the first assigned environment.
+
+If you reassign a host's content view environments using the web UI, the host will no longer be a multi-environment host.
+
+Assigning or registering a host to multiple environments requires the Allow multiple content views Setting to be on.
```
Comment (bangelic) [inline guides/doc-Release_Notes/topics/katello.adoc:43]:
```suggestion
Assigning or registering a host to multiple environments requires the *Allow multiple content views* setting to be on. 
```

**Comment by chris1984** [comment]
@ekohl @bangelic how does this look? I got all the issues addressed.

File: guides/doc-Release_Notes/master.adoc [hunk 16]
```diff
@@ -13,6 +13,7 @@ include::topics/katello.adoc[leveloffset=+1]
 endif::[]
 
 // Start inserting specific x.y.z releases here
+include::topics/katello-4.14.0.adoc[leveloffset=+1]
```
Comment (ekohl) [inline guides/doc-Release_Notes/master.adoc:16]: Looks like we need to enhance the procedure to only show it for Katello
```suggestion
ifdef::katello[]
include::topics/katello-4.14.0.adoc[leveloffset=+1]
endif::[]
```

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 17]
```diff
@@ -4,12 +4,65 @@
...
+* Content source
+* Content view environments - change the content view and lifecycle environment of hosts
+
+*Single-host actions* - In addition to actions added by Foreman, you can click the vertical ellipsis in the rightmost table column (without selecting the host) to change content view environments on a single host.
+
+*Column selector* - You can now click Manage Columns and customize columns displayed in the table.
+The column options are the same as those on the legacy All Hosts page. 
+Any changes made in either the new experimental *All Hosts* page and the legacy *All Hosts* page reflect on each other.
+
+=== Assigning multiple content view environments to hosts
```
Comment (ekohl) [inline guides/doc-Release_Notes/topics/katello.adoc:26]: I think this reads better, but I'm looking at @bangelic for a definitive answer
```suggestion
=== Assign multiple content view environments to hosts
```

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 18]
```diff
@@ -4,12 +4,65 @@
...
+
+You can now assign multiple content view environments to a host through subscription-manager or Hammer. A multi-environment host will have access to enabled repositories from all of its assigned content view environments. This strategy may be more convenient than composite content views for instances where frequent publishes are required.
+
+Through hammer, pass the `content-view-environments` param to `hammer host update` in a comma-separated list in the format `<lifecycle_environment_label>/<content_view_label>` . Ex `hammer host update --id xxx --content-view-environments lce1/cv1,lce2/cv2`
+
+On the host, run `subscription-manager environments --set` and pass in a comma-separated list. 
+Ex. `subscription-manager environments --set lce1/cv1,lce2/cv2` .
+To see the list of possible content view environments, use subscription-manager environments --list
+To see the currently assigned content view environment(s), use subscription-manager environments --list-enabled
```
Comment (ekohl) [inline guides/doc-Release_Notes/topics/katello.adoc:35]:
```suggestion
To see the list of possible content view environments, use `subscription-manager environments --list`
To see the currently assigned content view environment(s), use `subscription-manager environments --list-enabled`
```

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 19]
```diff
@@ -4,12 +4,65 @@
...
+On {SmartProxyServer}, the deprecated port 8443 has been disabled by default on new installations and upgrades.
+Clients are expected to be using port 443 when communicating through a {SmartProxyServer}.
+If there are older clients still using port 8443, they should be re-configured either by using remote execution, the katello-ca-consumer RPM, or updating `/etc/rhsm/rhsm.conf` using `subscription-manager --port 443`.
+
+For users that temporarily need access to port 8443 again, the installer can be executed with `{foreman-installer} --foreman-proxy-content-reverse-proxy true`.
+This should only be used as a temporary solution as port 8443 will be removed permanently in the next release.
+
+=== Pulp performance issue
+
+Extra Pulpcore indexes to address a https://github.com/pulp/pulpcore/issues/5367[performance issue] will be added during the upgrade. If you've manually created these indexes, please remove them before proceeding with the upgrade to ensure that the migration runs successfully.
```
Comment (ekohl) [inline guides/doc-Release_Notes/topics/katello.adoc:65]:
```suggestion
Extra Pulpcore indexes to address a https://github.com/pulp/pulpcore/issues/5367[performance issue] will be added during the upgrade.
If you've manually created these indexes, please remove them before proceeding with the upgrade to ensure that the migration runs successfully.
```

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 20]
```diff
@@ -4,12 +4,65 @@
...
+However, multi-environment hosts will now be displayed in the web UI on the new All Hosts page and new host details page.
+
+On the new All Hosts page, there is a new 'Content view environments' column available to add. This will display all assigned environments for a host. The existing 'Content view' and 'Lifecycle environment' columns will continue to work, but will only display the first assigned environment.
+
+If you reassign a host's content view environments using the web UI, the host will no longer be a multi-environment host.
+
+Assigning or registering a host to multiple environments requires the *Allow multiple content views* setting to be on. 
+If you turn this setting off, existing multi-environment hosts will continue to work, but you will not be able to assign multiple environments to hosts going forward.
+
+You can also assign multiple environments at registration using subscription-manager register --environments, but global registration and activation keys are not yet supported.
```
Comment (ekohl) [inline guides/doc-Release_Notes/topics/katello.adoc:47]:
```suggestion
You can also assign multiple environments at registration using `subscription-manager register --environments`, but global registration and activation keys are not yet supported.
```

**Review by ekohl — CHANGES_REQUESTED** [review]

File: guides/doc-Release_Notes/topics/katello.adoc (hunk 17 above)
Comment (bangelic) [reply guides/doc-Release_Notes/topics/katello.adoc]: I agree with @ekohl

**Comment by chris1984** [comment]
@ekohl committed all your suggestions and rebased to one commit, when you have time to check again. Hopefully, third time is a charm :)

**Review by ekohl — APPROVED** [review]

**Comment by ekohl** [comment]
@bangelic please merge if you think this is ready.

**Comment by bangelic** [comment]
@Lennonka or @asteflova can one of you merge this? It got the ACK from @ekohl. Thanks!

**Comment by ekohl** [comment]
Oh yes, I forgot you don't have merge permissions.

**Pull request merged**